pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -353,6 +367,47 @@ func (p *tinkerbellProvider) SetupAndValidateCreateCluster(ctx context.Context,
 	return nil
 }
 
+// setMachinesToPXEBoot iterates over all p.validator.hardwareConfig.Bmcs and instructs them to turn off
+// and boot from PXE and turn on.
+func (p *tinkerbellProvider) setMachinesToPXEBoot(ctx context.Context) error {
+	// We're reaching into an unexported member of p.validator because of the lazy loading we're doing with
+	// hardware configuration. This effectively defines a concrete tight coupling between the validator and the
+	// Tinkerbell construct that desperately needs teething apart.
+
+	secrets := make(map[string]corev1.Secret, len(p.validator.hardwareConfig.Secrets))
+	for _, secret := range p.validator.hardwareConfig.Secrets {
+		secrets[secret.Name] = secret
+	}
+
+	var errs []error
+	for _, bmc := range p.validator.hardwareConfig.Bmcs {
+		conf := pbnj.BmcSecretConfig{
+			Host:     bmc.Spec.Host,
+			Username: secrets[bmc.Spec.AuthSecretRef.Name].StringData["username"],
+			Password: secrets[bmc.Spec.AuthSecretRef.Name].StringData["password"],
+			Vendor:   bmc.Spec.Vendor,
+		}
+
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+
+		if err := p.pbnj.PowerOff(ctx, conf); err != nil {
+			errs = append(errs, err)
+		}
+
+		if err := p.pbnj.SetBootDevice(ctx, conf, pbnj.BootDevicePXE); err != nil {
+			errs = append(errs, err)
+		}
+
+		if err := p.pbnj.PowerOn(ctx, conf); err != nil {
+			errs = append(errs, err)
+		}
```

Comment:
why are we powering the machines on here? 
CAPT's pbnj should take care of this and that also won't power on all the machines, only the ones that are needed for the cluster

Reply:
Yea I realized that. Will remove once I can actually get a test functional.